### ISSUE TITLE
Android SDL2 2.30.0 regression, on screen keyboard popping up on resume activity

### ISSUE THREAD

**Issue by 312937** [libsdl-org/SDL#9202]
**Target**: Android
**SDL2 version**: 2.30.0 comparing to 2.28.5 and before
**Observed**:
 If **_any_** application goes to background and is re-activated (on resume) the on-screen keyboard pops up immediately even though there was/is no text input context  all.
Apparently:
SDL2-2.30.0/src/video/android/SDL_androidevents.c
```
            /* Make sure SW Keyboard is restored when an app becomes foreground */
            if (SDL_IsTextInputActive() &&
                SDL_GetHintBoolean(SDL_HINT_ENABLE_SCREEN_KEYBOARD, SDL_TRUE)) {
                Android_ShowScreenKeyboard(_this, Android_Window); /* Only showTextInput */
            }
```
is unfortunately not equivalent to:
SDL2-2.28.5/src/video/android/SDL_androidevents.c
```
            /* Make sure SW Keyboard is restored when an app becomes foreground */
            if (SDL_IsTextInputActive()) {
                Android_StartTextInput(_this); /* Only showTextInput */
            }
```
To reproduce:
1. Any application
2. Send to background
3. Resume (get into focus)

**Comment by slouken** [comment]
Fixed, thanks!

**Comment by slouken** [comment]
This fix is in the 2.30.1 release, out now.
